lib/improver/nowcasting/nowcasting.py

```diff
@@ -0,0 +1,196 @@
+
+    """Apply orographic enhancement to precipitation rate input, either to
+     add or subtract an orographic enhancement component."""
+
+    def __init__(self, operation):
+        """Initialise class.
+
+        Args:
+            operation (str):
+                Operation (+, add, -, subtract) to apply to the incoming cubes.
+
+        Raises:
+            ValueError: Operation not supported.
+
+        """
+        possible_operations = ['+', 'add', '-', 'subtract']
+
+        if operation in possible_operations:
+            self.operation = operation
+        else:
+            msg = ("Operation '{}' not supported for combining "
+                   "precipitation rate and "
+                   "orographic enhancement.".format(operation))
+            raise ValueError(msg)
+
+    def __repr__(self):
+        """Represent the configured plugin instance as a string."""
+        result = ('<ApplyOrographicEnhancement: operation: {}>')
+        return result.format(self.operation)
+
+    @staticmethod
+    def _extract_orographic_enhancement_cube(precip_cube, oe_cubes):
+        """Extract the orographic enhancement cube with the required time
+        coordinate.
+
+        Args:
+            precip_cube (iris.cube.Cube):
+                Cube containing the input precipitation fields.
+            oe_cubes (iris.cube.Cube or iris.cube.CubeList):
+                Cube or CubeList containing the orographic enhancement fields.
+
+        Returns:
+            oe_cube (iris.cube.Cube):
+                Cube containing the orographic enhancement fields at the
+                required time.
+
+        """
+        time_point, = iris_time_to_datetime(precip_cube.coord("time"))
+        oe_cube = extract_nearest_time_point(oe_cubes, time_point)
+        return oe_cube
+
+    def _apply_cube_combiner(self, precip_cube, oe_cube):
+        """Combine the precipitation rate cube and the orographic enhancement
+        cube.
+
+        Args:
+            precip_cube (iris.cube.Cube):
+                Cube containing the input precipitation field.
+            oe_cube (iris.cube.Cube):
+                Cube containing the orographic enhancement field.
+
+        Returns:
+            cube (iris.cube.Cube):
+                Cube containing the precipitation rate field modified by the
+                orographic enhancement cube.
+
+        """
+        # Ensure the orographic enhancement cube matches the
+        # dimensions of the precip_cube.
+        oe_cube = check_cube_coordinates(precip_cube, oe_cube)
+
+        # Ensure that orographic enhancement is in the units of the
+        # precipitation rate cube.
+        oe_cube.convert_units(precip_cube.units)
+
+        # Set orographic enhancement to be zero for points with a
+        # precipitation rate of < 1/32 mm/hr.
+        original_units = Unit("mm/hr")
+        threshold_in_cube_units = (
+            original_units.convert(1/32., precip_cube.units))
```

Comment:
1/32 is a magic number - please define this as a class member `self.min_precip_rate_mmh` or similar.

Reply:
I've made this a class member as suggested.